### DISCUSSION TITLE
handlling NaN values in netcdfs

### DISCUSSION THREAD

**Discussion by KayCritchell** [OpenDrift/opendrift#1013] [Q&A] [answered]
Hello,

First: thanks for making this project open accessible.

I'm attempting to use hydro outputs from the eReefs project (e.g. 'https://dapds00.nci.org.au/thredds/fileServer/fx3/gbr1_2.0/gbr1_simple_2019-03-19.nc') but these files contain NaN values in the land mask and so I get "ValueError: Points cannot contain NaN" when I try to load them into reader_netCDF_CF_generic. 
I cant find documentation on what the readers will accept. How should I be handling non-value cells?

full error:

Traceback (most recent call last):
  File "testingOutOpenDrift.py", line 16, in <module>
    r = reader_netCDF_CF_generic.Reader(
  File "\opendrift\readers\reader_netCDF_CF_generic.py", line 345, in __init__
    super().__init__()
  File "\opendrift\readers\basereader\structured.py", line 74, in __init__
    self.spl_x = LinearNDInterpolator(
  File "interpnd.pyx", line 280, in scipy.interpolate.interpnd.LinearNDInterpolator.__init__
  File "_qhull.pyx", line 1846, in scipy.spatial._qhull.Delaunay.__init__
  File "_qhull.pyx", line 285, in scipy.spatial._qhull._Qhull.__init__
ValueError: Points cannot contain NaN

testingOutOPenDrift.py contains:

```
from opendrift.readers import reader_netCDF_CF_generic
from opendrift.models.oceandrift import OceanDrift
from datetime import timedelta, datetime
import matplotlib.pyplot as plt
import numpy as np

o = OceanDrift(loglevel = 20)

#how to add multiple .nc files into one reader
r = reader_netCDF_CF_generic.Reader(
	r'Hydro/ereefs/gbr1_2019-03-19.nc')

print(r)
```

**Comment by knutfrode** [comment, marked as answer]
Hi Kay,

I believe the problem is not that landmask contains NaN, but rather that longitude and latitude arrays contain NaN, so that the interpolator from (i,j) <-> (lon,lat) cannot be created.
However, lokking at your file with ncview, there seem to be other issues, e.g. u and v (current components) seem to be vanishing except for a very small area at the bottom. So perhaps this file has become corrupted somehow?

**Reply by KayCritchell** [reply]
ahh thanks for your help! I clearly didn't read the error message carefully. I will chat with the data owners to clarify and sort it out on the data end. Thanks again
